project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -60,10 +69,61 @@ public void updateUser(User user) {
     return Mutation.newUpdateBuilder(USER_TABLE);
   }
 
-  private static List<Mutation> getMutationsFromBuilder(Mutation.WriteBuilder builder, User user) {
+  /**
+   * Given an event, insert or update a row with all available fields into the DB
+   *
+   * @param event the event to be inserted or updated; event's ID field should not exist in DB
+   */
+  public void insertorUpdateEvent(Event event) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
+
+    List<Mutation> mutations =
+        getEventMutationsFromBuilder(Mutation.newInsertOrUpdateBuilder(EVENT_TABLE), event);
+    dbClient.write(mutations);
+    spanner.close();
+  }
+
+  /** Returns Event by ID from DB
+   * 
+   * @param eventId ID of event to be returned
+  */
+  public Optional<Event> getEventById(String eventId) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
```

Comment:
Use injection for DatabaseWrapper or share the initialization for all the operations.

Reply:
done